httpmw/logging_mw.go

```diff
@@ -0,0 +1,70 @@
+package httpmw
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+type Logger interface {
+	Info(msg string, keysAndValues ...interface{})
+	Error(msg string, keysAndValues ...interface{})
+}
+
+func NewLogging(logger Logger) mux.MiddlewareFunc {
```

Comment:
I didn't find examples how this part works. If you have links, please add as comment.

Reply:
The mux.MiddlewareFunc is a type defined in gorilla/mux which is used to wrap the existing http handler to add logging functionality in this case. 

Here is an example of usage in Fleet: https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=/fleet/pkg/api/middleware/logging_mw.go&version=GBmaster&line=21&lineEnd=22&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Added comment in the code with link to gorilla/mux doc.